### ISSUE TITLE
Document process for embedding benchmark in external project

### ISSUE THREAD

**Issue by dominichamon** [google/benchmark#788]
There are a number of Stack Overflow, mailing lists, and IRC enquiries about embedding Benchmark in other projects. Using cmake this should be trivial, but it seems on Windows in particular there are some confusing elements.

Specifically the docs should mention the use of `BENCHMARK_DOWNLOAD_DEPENDENCIES`, static vs dynamic library generation, vcpkg/installation vs building from head.

**Comment by SayantanMajumdar** [comment]
I have written some additional build documentation, in a new branch in my fork. It demonstrates the use of `BENCHMARK_DOWNLOAD_DEPENDENCIES`  and vcpkg/installation vs building from head(I think it does). The documentation is in a new file `docs/Additional_Build_Documentation.md` and I have added a link to it from the `README.md` file. I am quite new to using Benchmark, I am not sure about the quality of the doc that I have written. Should I make a pull request? I am sure it would require some improvement.

**Comment by LebedevRI** [comment]
Linking to googletest docs could be good too, it's basically the same with maybe a few additional specifics.

**Comment by SayantanMajumdar** [comment]
yes, I mentioned that it is similar but did not link it because it would make this documentation dependent on the documentation location of googletest. But I can surely change it.
